source/Actions/Fido.Gui.Fmx.Action.Anon.pas

```diff
@@ -0,0 +1,133 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *)
+
+ unit Fido.Gui.Fmx.Action.Anon;
+
+interface
+
+uses
+  System.SysUtils,
+  System.Classes,
+  System.Rtti,
+  System.TypInfo,
+  System.AnsiStrings,
+  Fmx.ActnList,
+  Fmx.Controls,
+  Fmx.Types,
+
+  Fido.Gui.Types,
+  Fido.Gui.Fmx.NotifyEvent.Delegated;
+
+type
+  AnonAction = class
+  private type
+    TExFmxObject = class(TFmxObject)
+    published
+      property Action;
+    end;
+  public
+    class procedure Setup(const Owner: TComponent; const Control: TFmxObject; const OnExecuteProc: TProc<TObject>; const OnExecuteProcOriginalEventExecutionType: TOriginalEventExecutionType = oeetBefore;
+      const OnChangeProc: TProc<TObject> = nil; const OnChangeProcOriginalEventExecutionType: TOriginalEventExecutionType = oeetBefore; const OnStateChangeProc: TProc<TObject> = nil; const OnStateChangeProcOriginalEventExecutionType: TOriginalEventExecutionType = oeetBefore; const OnUpdateProc: TProc<TObject> = nil; const OnUpdateProcOriginalEventExecutionType: TOriginalEventExecutionType = oeetBefore); overload; static;
+
+    class procedure Setup<T>(const Owner: TComponent; const Control: TFmxObject; const Executer: T; const ExecuterMethodName: string;
+      const OnExecuteProcOriginalEventExecutionType: TOriginalEventExecutionType = oeetBefore); overload; static;
+  end;
+
+implementation
+
+{ AnonAction }
+
+class procedure AnonAction.Setup(
+  const Owner: TComponent;
+  const Control: TFmxObject;
+  const OnExecuteProc: TProc<TObject>;
+  const OnExecuteProcOriginalEventExecutionType: TOriginalEventExecutionType;
+  const OnChangeProc: TProc<TObject>;
+  const OnChangeProcOriginalEventExecutionType: TOriginalEventExecutionType;
+  const OnStateChangeProc: TProc<TObject>;
+  const OnStateChangeProcOriginalEventExecutionType: TOriginalEventExecutionType;
+  const OnUpdateProc: TProc<TObject>;
+  const OnUpdateProcOriginalEventExecutionType: TOriginalEventExecutionType);
+var
+  Action: TAction;
+  Ctx: TRttiContext;
+  ActionProp: TRttiProperty;
+  CaptionProp: TRttiProperty;
+begin
+  Action := TAction.Create(Owner);
+
+  Ctx := TRttiContext.Create;
+  ActionProp := Ctx.GetType(Control.ClassType).GetProperty('Action');
+  if not Assigned(ActionProp) then
+    Exit;
```

Comment:
Should this function just exit silently? 
How does the user (in this case the library itself as far as I can see) get feedback on a possible error on the users part?